tests/e2e-pw/specs/merchant/merchant-orders-status-change.spec.ts

```diff
@@ -0,0 +1,177 @@
+/**
+ * External dependencies
+ */
+import { test, expect, Page } from '@playwright/test';
+
+/**
+ * Internal dependencies
+ */
+import { getMerchant, getShopper, isUIUnblocked } from '../../utils/helpers';
+import { placeOrderWithOptions } from '../../utils/shopper';
+import * as navigation from '../../utils/merchant-navigation';
+
+const orderStatusDropdownSelector = 'select[name="order_status"]';
+const cancelModalSelector = 'div.wcpay-confirmation-modal';
+const refundModalSelector = 'div.refund-confirmation-modal';
+const refundCancelSelector =
+	'.refund-confirmation-modal .wcpay-confirmation-modal__footer .is-secondary';
+const refundConfirmSelector =
+	'.refund-confirmation-modal .wcpay-confirmation-modal__footer .is-primary';
+const selectedOrderStatusSelector = '.wc-order-status > span';
+const orderPriceSelector =
+	'#woocommerce-order-items .total .woocommerce-Price-amount';
+
+const saveOrder = async ( page: Page ) => {
+	await page.locator( '.save_order' ).click();
+	await page.waitForLoadState( 'networkidle' );
+};
+
+const verifyOrderStatus = async ( page: Page, status: string ) => {
+	const selectedOrderStatus = await page.$( selectedOrderStatusSelector );
+	await expect(
+		selectedOrderStatus.evaluate( ( el ) => el.textContent )
+	).resolves.toBe( status );
+};
+
+test.describe( 'Order > Status Change', () => {
+	let merchantPage: Page;
+	let shopperPage: Page;
+	let orderId: string;
+
+	test.beforeAll( async ( { browser } ) => {
+		merchantPage = ( await getMerchant( browser ) ).merchantPage;
+		shopperPage = ( await getShopper( browser ) ).shopperPage;
+	} );
+
+	test.describe( 'Change Status of order to Cancelled', () => {
+		test.beforeAll( async () => {
+			orderId = await placeOrderWithOptions( shopperPage );
+			await navigation.goToOrder( merchantPage, orderId );
+		} );
+
+		test( 'Show Cancel Confirmation modal, do not change status if Do Nothing selected', async () => {
+			// Select cancel from the order status dropdown.
+			await merchantPage.selectOption(
+				orderStatusDropdownSelector,
+				'Cancelled'
+			);
+
+			// Verify the confirmation modal shows.
+			await merchantPage.waitForSelector( cancelModalSelector, {
+				state: 'visible',
+			} );
```

Comment:
In the [doc for `waitForSelector`](https://playwright.dev/docs/api/class-elementhandle#element-handle-wait-for-selector) it says:

> Use web assertions that assert visibility or a locator-based [locator.waitFor()](https://playwright.dev/docs/api/class-locator#locator-wait-for) instead.

Would we want to swap these out? The code would be a little more complex from what I see, but it may be a little more future proof, unless I am missing something.

Reply:
Great call! I've updated the code to use locator.waitFor() instead on [70add24](https://github.com/Automattic/woocommerce-payments/pull/10243/commits/70add24e7afa9e4368f0ec210b4d9acf63c62c0f). Thanks for raising that!